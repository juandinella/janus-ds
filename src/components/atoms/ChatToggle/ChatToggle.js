import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './ChatToggle.styles'
import ScreenReaderText from '../ScreenReaderText'

/**
 * ChatToggle
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledChatToggle = styled.button.withConfig({ shouldForwardProp })`
  ${styles}
`

const ChatToggle = ({
  id,
  disabled = false,
  isClicked = false,
  onClick = () => null,
  backgroundColor = '#171717',
  color = '#FFFFFF',
}) => {
  const handleClick = () => {
    if (onClick) {
      onClick()
    }
  }

  return (
    <StyledChatToggle
      id={id}
      data-testid={id}
      onClick={handleClick}
      disabled={disabled}
      backgroundColor={backgroundColor}
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="20"
        height="20"
        fill="none"
        shapeRendering="geometricPrecision"
        vectorEffect="non-scaling-stroke"
      >
        {isClicked ? (
          <>
            <path
              fill={color}
              fillRule="evenodd"
              d="M15.59 4.41a.833.833 0 010 1.18l-10 10a.833.833 0 11-1.18-1.18l10-10a.833.833 0 011.18 0z"
              clipRule="evenodd"
            />
            <path
              fill={color}
              fillRule="evenodd"
              d="M4.41 4.41a.833.833 0 011.18 0l10 10a.833.833 0 11-1.18 1.18l-10-10a.833.833 0 010-1.18z"
              clipRule="evenodd"
            />
          </>
        ) : (
          <path
            fill={color}
            fillRule="evenodd"
            d="M4.167 3.333a.833.833 0 00-.834.834v11.321l1.911-1.91a.833.833 0 01.59-.245h10a.833.833 0 00.833-.833V4.167a.833.833 0 00-.834-.834H4.167zM2.399 2.4a2.5 2.5 0 011.768-.732h11.666a2.5 2.5 0 012.5 2.5V12.5a2.5 2.5 0 01-2.5 2.5H6.18l-3.09 3.09a.833.833 0 01-1.422-.59V4.167A2.5 2.5 0 012.4 2.399z"
            clipRule="evenodd"
          />
        )}
      </svg>
      <ScreenReaderText>
        {isClicked ? 'Close Chat' : 'Open Chat'}
      </ScreenReaderText>
    </StyledChatToggle>
  )
}

ChatToggle.displayName = 'ChatToggle'

ChatToggle.propTypes = {
  id: PropTypes.string,
  disabled: PropTypes.bool,
  onClick: PropTypes.func,
  isClicked: PropTypes.bool,
  backgroundColor: PropTypes.string,
  color: PropTypes.string,
}

export default ChatToggle
