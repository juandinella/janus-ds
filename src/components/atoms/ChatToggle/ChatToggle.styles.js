import { css } from 'styled-components'

export default css`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 50px;
  height: 50px;
  border-radius: var(--border-radius-round);
  box-shadow: var(--box-shadow-main);
  transition: transform 250ms cubic-bezier(0.33, 0, 0, 1);
  transform: scale3d(1, 1, 1);
  backface-visibility: hidden;
  will-change: transform;
  transform-origin: center center;
  margin-left: auto;

  &:hover {
    transform: scale3d(1.1, 1.1, 1.1);
  }

  &:focus-visible {
    box-shadow:
      0 0 0 2px white,
      0 0 0 4px #e5e5e5;
  }

  ${(props) =>
    props.backgroundColor &&
    css`
      background-color: ${props.backgroundColor};
    `}
`
