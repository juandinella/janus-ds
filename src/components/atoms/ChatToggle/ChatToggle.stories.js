import ChatToggle from './ChatToggle'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(ChatToggle)

export default {
  title: 'Atoms/ChatToggle',
  component: ChatToggle,
  args: {
    color: '#FFFFFF',
    backgroundColor: '#171717',
  },
  argTypes: {
    color: {
      control: { type: 'color' },
    },
    backgroundColor: {
      control: { type: 'color' },
    },
  },
}

export const Default = Template.bind({})
