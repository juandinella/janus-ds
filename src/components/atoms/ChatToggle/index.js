import ChatToggle from './ChatToggle'

export default ChatToggle
