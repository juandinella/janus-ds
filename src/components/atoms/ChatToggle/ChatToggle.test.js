import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import React from 'react'
import ChatToggle from './ChatToggle'

initStoryshots({
  storyKindRegex: /^Atoms\/ChatToggle$/,
})

const componentProps = {
  id: 'unique_id',
  onClick: jest.fn(),
}
afterEach(cleanup)
beforeEach(() => {
  componentProps.onClick.mockClear()
})

describe('<ChatToggle />', () => {
  test('should render component correctly', () => {
    render(<ChatToggle {...componentProps} />)
    const btn = screen.getByTestId(componentProps.id)
    expect(btn).toBeInTheDocument()
  })
  test('should call function on click', () => {
    const { getByTestId } = render(<ChatToggle {...componentProps} />)
    const button = getByTestId(componentProps.id)
    fireEvent.click(button)
    fireEvent.click(button)
    expect(componentProps.onClick).toHaveBeenCalledTimes(2)
  })
})
