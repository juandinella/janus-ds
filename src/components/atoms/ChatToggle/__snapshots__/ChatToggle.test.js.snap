// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/ChatToggle Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <button
    className="sc-nwx0we-0 gnFNja"
    disabled={false}
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="20"
      shapeRendering="geometricPrecision"
      vectorEffect="non-scaling-stroke"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M4.167 3.333a.833.833 0 00-.834.834v11.321l1.911-1.91a.833.833 0 01.59-.245h10a.833.833 0 00.833-.833V4.167a.833.833 0 00-.834-.834H4.167zM2.399 2.4a2.5 2.5 0 011.768-.732h11.666a2.5 2.5 0 012.5 2.5V12.5a2.5 2.5 0 01-2.5 2.5H6.18l-3.09 3.09a.833.833 0 01-1.422-.59V4.167A2.5 2.5 0 012.4 2.399z"
        fill="#FFFFFF"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="sc-90c4yq-0 bmmsAZ"
    >
      Open Chat
    </span>
  </button>
</div>
`;
